src/contexts/SendContext.tsx

```diff
@@ -0,0 +1,210 @@
+    const newAssetAmounts = [...assetAmounts]
+
+    if (existingAmountIndex !== -1) {
+      amount
+        ? newAssetAmounts.splice(existingAmountIndex, 1, { id: assetId, amount })
+        : newAssetAmounts.splice(existingAmountIndex, 1)
+    } else {
+      newAssetAmounts.push({ id: assetId, amount })
+    }
+
+    setAssetAmounts(newAssetAmounts)
+  }
+
+  const buildConsolidationTransactions = useCallback(async () => {
+    if (!address) return
+
+    try {
+      const data = await buildSweepTransactions(address, address.hash)
+      setUnsignedTxData(data)
+    } catch (e) {
+      Toast.show(getHumanReadableError(e, 'Error while building the transaction'))
+    }
+  }, [address, setUnsignedTxData])
+
+  const buildTransaction = useCallback(
+    async (callbacks: BuildTransactionCallbacks) => {
+      if (!address || !toAddress) return
+
+      try {
+        const data = await buildUnsignedTransactions(address, toAddress, assetAmounts)
+        setUnsignedTxData(data)
+
+        callbacks.onBuildSuccess()
+      } catch (e) {
+        // TODO: When API error codes are available, replace this substring check with a proper error code check
+        const { error } = e as APIError
+        if (error?.detail && (error.detail.includes('consolidating') || error.detail.includes('consolidate'))) {
+          setConsolidationRequired(true)
+          setIsConsolidateModalVisible(true)
+          setOnSendSuccessCallback(() => callbacks.onConsolidationSuccess)
+          await buildConsolidationTransactions()
+        } else {
+          Toast.show(getHumanReadableError(e, 'Error while building the transaction'))
+        }
+      }
+    },
+    [address, assetAmounts, buildConsolidationTransactions, toAddress]
+  )
+
+  const sendTransaction = useCallback(
+    async (onSendSuccess: () => void) => {
+      if (!address || !toAddress) return
+
+      const { attoAlphAmount, tokens } = getTransactionAssetAmounts(assetAmounts)
+
+      try {
+        for (const { txId, unsignedTx } of unsignedTxData.unsignedTxs) {
+          const data = await signAndSendTransaction(address, txId, unsignedTx)
+
+          dispatch(
+            transactionSent({
+              hash: data.txId,
+              fromAddress: address.hash,
+              toAddress: consolidationRequired ? address.hash : toAddress,
+              amount: attoAlphAmount,
+              tokens,
+              timestamp: new Date().getTime(),
+              status: 'pending'
+            })
+          )
+        }
+
+        onSendSuccess()
+      } catch (e) {
+        Toast.show(getHumanReadableError(e, 'Could not send transaction'))
+      }
+    },
+    [address, assetAmounts, consolidationRequired, dispatch, toAddress, unsignedTxData.unsignedTxs]
+  )
+
+  const authenticateAndSend = useCallback(
+    async (onSendSuccess: () => void) => {
+      if (requiresAuth) {
+        setOnSendSuccessCallback(() => onSendSuccess)
```

Comment:
This little `() =>` was the solution to the problem I mentioned on Slack today ;)